### DISCUSSION TITLE
灵墨 OS 2.0 Hydrogen 发布啦！

### DISCUSSION THREAD

**Discussion by elysia-best** [LingmoOS/Community#20] [社区通知]
我们很高兴地宣布，灵墨 OS 2.0 终于发布了！
您可以在 [Sourceforge](https://sourceforge.net/projects/lingmo-os/files/latest/download) 上下载最新版本。
以下是此版本的发行说明。

<h1 id="lingmo-os-2-0-hydrogen-release-notes" _msttexthash="42759496" _msthash="0">Lingmo OS 2.0 Hydrogen 发行说明</h1>
<p _msttexthash="506710737" _msthash="1"><strong _istranslated="1">注意</strong>：<em _istranslated="1">2.0 （Hydrogen） 是过渡版本，所以没有增加太多功能，请等待一个月后发布的 3.0 （Helium）。</em></p>
<p><strong _msttexthash="20276633" _msthash="2">生命周期：</strong></p>
<ul>
<li _msttexthash="800551869" _msthash="3">2.0 （Hydrogen） 是一个过渡版本，因此它的支持期只有一个月，但我们建议用户安装它，因为它可以接受此版本之后的所有更新。</li>
</ul>
<p><strong _msttexthash="11495653" _msthash="4">新增功能</strong></p>
<ul>
<li _msttexthash="23257117" _msthash="5">内核版本升级到 <strong _istranslated="1"><em _istranslated="1">6.10.9</em></strong></li>
<li _msttexthash="139164129" _msthash="6">合并来自上游 <strong _istranslated="1">Debian GNU/Linux</strong> 的安全更新和最新软件包</li>
<li _msttexthash="42615560" _msthash="7">更好的视觉效果和体验</li>
<li _msttexthash="33165483" _msthash="8">支持最新的固件版本</li>
<li _msttexthash="24836292" _msthash="9">系统版本升级到 <strong _istranslated="1"><em _istranslated="1">2.0</em></strong></li>
<li><font _mstmutation="1" _msttexthash="13815022" _msthash="10">代号升级为 <strong _mstmutation="1" _istranslated="1"><em _istranslated="1">Hydrogen</em></strong></font> </li>
</ul>
<p><strong _msttexthash="18864963" _msthash="11">修复的 bug：</strong></p>
<ul>
<li _msttexthash="48871277" _msthash="12">修复了无法更新包的问题</li>
<li _msttexthash="69456088" _msthash="13">修复了无法使用软件商店的问题</li>
<li _msttexthash="59235644" _msthash="14">修复了某些页面的崩溃问题</li>
</ul>
<p><strong _msttexthash="27558791" _msthash="15">下一步计划</strong></p>
<ul>
<li _msttexthash="39628511" _msthash="16">完整的文件管理器功能</li>
<li _msttexthash="27097668" _msthash="17">系统资源管理器</li>
<li _msttexthash="25940382" _msthash="18">系统图片浏览器</li>
<li _msttexthash="36231819" _msthash="19">系统软件包安装程序</li>
<li _msttexthash="35986912" _msthash="20">完整的系统更新功能</li>
<li _msttexthash="26276471" _msthash="21">完整的桌面功能</li>
<li _msttexthash="16930537" _msthash="22">桌面小部件</li>
<li _msttexthash="9717383" _msthash="23">工作区域</li>
<li _msttexthash="59251413" _msthash="24">完整的启动功能和全屏/半屏切换</li>
<li _msttexthash="152765236" _msthash="25">将 U 盘图标拖拽到回收站图标上，弹出设备 （？</li>
</ul>
<p _msttexthash="23529909" _msthash="26"><strong _istranslated="1">Lingmo OS 开发团队</strong> <a href="lingmo.org" _istranslated="1">lingmo.org</a></p>

<yd-mg-icon style="position: fixed; z-index: 2147483647;"></yd-mg-icon><yd-mg-block-icon></yd-mg-block-icon><yd-image-ocr style="z-index: 2147483647;"></yd-image-ocr>

**Comment by samwhelp** [comment]
分享我個人在「Lingmo OS 2.0 Hydrogen」的「慣用設定」。

相關的設定檔，有放在「[GitHub](https://github.com/samwhelp/lingmo-adjustment/tree/main/prototype/main/lingmo-config/locale/en_us/Lingmo-Dark/asset/overlay/etc/skel/.config)」上，

只要複製到「~/.config」這個資料夾，然後先登出，再登入，就會生效。

這個設定，有如下兩個重點

* 開啟了「[wobblywindows](https://github.com/samwhelp/lingmo-adjustment/blob/main/prototype/main/lingmo-config/locale/en_us/Lingmo-Dark/asset/overlay/etc/skel/.config/kwinrc#L57)」的效果，在「拖拉移動視窗」時，就可以看到該效果。
* 我個人慣用的「按鍵綁定」。

關於有哪些「按鍵綁定」，我有撰寫了文件如下

| 文件 |
| ----- |
| [鍵盤按鍵綁定一覽表](https://samwhelp.github.io/note-about-lingmo/read/cheatsheet/keybind.html) |
| [鍵盤按鍵綁定設定說明](https://samwhelp.github.io/note-about-lingmo/read/config/keybind.html) |


以上分享，報告完畢

:-)

**Comment by samwhelp** [comment]
# 在「Lingmo」環境，如何透過「Variety」設定桌面圖片




## 設定桌面圖片基礎

> 做了點探索，了解到在「Lingmo」環境，可以透過「Dbus」來設定「桌面圖片」。


以下是參考指令


## dbus-wallpaper-set

執行下面指令，將「Lingmo」的「桌面圖片」設定為「/usr/share/backgrounds/lingmoos/grass1.jpg」。

``` sh
qdbus com.lingmo.Settings /Theme com.lingmo.Theme.setWallpaper "/usr/share/backgrounds/lingmoos/grass1.jpg"
```




## dbus-wallpaper-get

執行下面指令，獲得「Lingmo」目前設定的「桌面圖片路徑」。

``` sh
qdbus com.lingmo.Settings /Theme com.lingmo.Theme.wallpaper
```

顯示

```
/usr/share/backgrounds/lingmoos/grass1.jpg
```




## Variety 調整說明

因為目前「[Variety](https://github.com/varietywalls/variety)」，並沒有支援到「Lingmo」，

所以我們必須做些許的微調，才能讓「Variety」在「Lingmo」環境有作用。

應用上面提到的基礎，透過「Dbus」來設定「桌面圖片」，

我們需要稍微修改「`~/.config/variety/scripts/get_wallpaper`」和「`~/.config/variety/scripts/set_wallpaper`」這兩個腳本，就可以讓「Variety」在「Lingmo」環境正常運作。




## 修改「get_wallpaper」

> 加入程式碼片段: [~/.config/variety/scripts/get_wallpaper](https://github.com/samwhelp/lingmo-adjustment/blob/main/prototype/main/tool-config/part/variety/asset/overlay/etc/skel/.config/variety/scripts/get_wallpaper#L17-L20)

``` sh
# Lingmo OS
elif [ "$XDG_CURRENT_DESKTOP" == "Lingmo" ]; then

        qdbus com.lingmo.Settings /Theme com.lingmo.Theme.wallpaper
```




## 修改「set_wallpaper」

> 加入程式碼片段: [~/.config/variety/scripts/set_wallpaper](https://github.com/samwhelp/lingmo-adjustment/blob/main/prototype/main/tool-config/part/variety/asset/overlay/etc/skel/.config/variety/scripts/set_wallpaper#L208-L213)

``` sh
# Lingmo OS
if [ "$XDG_CURRENT_DESKTOP" == "Lingmo" ]; then

    qdbus com.lingmo.Settings /Theme com.lingmo.Theme.setWallpaper "$WP" 2> /dev/null

fi
```




## 注意事項

因為「variety」自動產生的「~/.config/autostart/variety.desktop」

裡面的「`Exec=`」那一行的寫法，在「Lingmo」沒有作用。

所以我們要將它稍微修改簡化如下，才能在登入系統後，自動執行「variety」。

``` ini
Exec=variety
```

完整內容，請參考我修改後的「[~/.config/autostart/variety.desktop](https://github.com/samwhelp/lingmo-adjustment/blob/main/prototype/main/tool-config/part/variety/asset/overlay/etc/skel/.config/autostart/variety.desktop#L6)」。




## 設定腳本

> 以下是完整的設定腳本

| 設定腳本 |
| --------|
| tool-config / [variety](https://github.com/samwhelp/lingmo-adjustment/tree/main/prototype/main/tool-config/part/variety) |




## 相關連結

* [在「Lingmo」環境，透過「Dbus」設定「桌面圖片」](https://samwhelp.github.io/note-about-lingmo/read/howto/set-wallpaper-by-dbus.html)
* [在「Lingmo」環境，透過「Variety」設定「桌面圖片」](https://samwhelp.github.io/note-about-lingmo/read/howto/set-wallpaper-by-variety.html)

**Reply by samwhelp** [reply]
之前有針對這個功能在「Variety」發了個「[Pull Request #720  Support Lingmo OS ](https://github.com/varietywalls/variety/pull/720)」。

剛剛收到通知，已經被「Merged」。

* [get_wallpaper](https://github.com/varietywalls/variety/blob/master/data/scripts/get_wallpaper#L17-L19)
* [set_wallpaper](https://github.com/varietywalls/variety/blob/master/data/scripts/set_wallpaper#L212-L215)

以上，報告完畢

**Reply by elysia-best** [reply]
> 之前有針對這個功能在「Variety」發了個「[Pull Request #720 Support Lingmo OS ](https://github.com/varietywalls/variety/pull/720)」。
> 
> 剛剛收到通知，已經被「Merged」。
> 
> * [get_wallpaper](https://github.com/varietywalls/variety/blob/master/data/scripts/get_wallpaper#L17-L19)
> * [set_wallpaper](https://github.com/varietywalls/variety/blob/master/data/scripts/set_wallpaper#L212-L215)
> 
> 以上，報告完畢

感谢Thanks♪(･ω･)ﾉ

**Comment by xmlijian** [comment]
3.0还发布不？等着呢，
